asyncapi/asgardeo/data_types.bal

```diff
@@ -92,6 +92,15 @@ public type LoginSuccessEvent record {
     GenericSecurityData securityData?;
 };
 
+# Login Failed event
+#
+# + eventData - Event data  
+# + securityData - Event related security data
+public type LoginFailedEvent record {
+    LoginFailedData eventData?;
+    GenericSecurityData securityData?;
```

Comment:
Seems these fields are optional. Make sure they aren't nillable. If they are nillable please handle it too